File: move-zeroes/move_zeroes.go
func moveZeroes(nums []int)  { 
    n := 0
    for i := 0; i < len(nums); i ++ {
        if nums[i] != 0 {
            nums[i], nums[n] = nums[n], nums[i]
            n += 1
        }
    }
    // for i, v := range nums {
    //     if v != 0 {
    //         nums[i], nums[n] = nums[n], nums[i]
    //         n += 1
    //     }
    // }
}

File: move-zeroes/move_zeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        n = 0 
        for i in range(len(nums)):
            if nums[i]:
                nums[i],nums[n] = nums[n],nums[i]
                n += 1